connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -43,10 +43,38 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeoutException;
 
-public class RestClient {
+public class RestClient implements AutoCloseable {
     private static final Logger log = LoggerFactory.getLogger(RestClient.class);
     private static final ObjectMapper JSON_SERDE = new ObjectMapper();
 
+    private final HttpClient client;
+    public RestClient(WorkerConfig config) {
+        client = new HttpClient(SSLUtils.createClientSideSslContextFactory(config));
+
+        client.setFollowRedirects(false);
+
+        try {
+            client.start();
+        } catch (Exception e) {
+            log.error("Failed to start RestClient: ", e);
+            throw new ConnectRestException(Response.Status.INTERNAL_SERVER_ERROR, "Failed to start RestClient: " + e.getMessage(), e);
+        }
+    }
+
+    // VisibleForTesting
+    RestClient(HttpClient client) {
+        this.client = client;
+    }
+
+    @Override
+    public void close() {
+        try {
+            client.stop();
+        } catch (Exception e) {
+            log.error("Failed to stop HTTP client", e);
+        }
```

Comment:
The signature of `Autocloseable::close` allows us to add a `throws Exception` clause to the method signature, and (AFAICT) the only non-testing call site for this method wraps it in `Utils::closeQuietly`.

Any reason to wrap the exception here, or can we just throw it to the caller?